src/main/java/org/jabref/gui/PreviewPanel.java

```diff
@@ -51,134 +39,103 @@
-
-        final String copy = "copy";
-        getInputMap(JComponent.WHEN_FOCUSED).put(Globals.getKeyPrefs().getKey(KeyBinding.COPY_PREVIEW), copy);
-        actionMap.put(copy, this.copyPreviewAction);
-    }
-
-    private JPopupMenu createPopupMenu() {
-        JPopupMenu menu = new JPopupMenu();
-        menu.add(this.printAction);
-        menu.add(this.copyPreviewAction);
-        this.basePanel.ifPresent(p -> menu.add(p.frame().getNextPreviewStyleAction()));
-        this.basePanel.ifPresent(p -> menu.add(p.frame().getPreviousPreviewStyleAction()));
-        return menu;
-    }
-
-    private void createPreviewPane() {
-        previewPane = new JEditorPaneWithHighlighting() {
-            @Override
-            public Dimension getPreferredScrollableViewportSize() {
-                return getPreferredSize();
-            }
-
-        };
-        previewPane.setMargin(new Insets(3, 3, 3, 3));
-
-        previewPane.setComponentPopupMenu(createPopupMenu());
-
-        previewPane.setEditable(false);
-        previewPane.setDragEnabled(true); // this has an effect only, if no custom transfer handler is registered. We keep the statement if the transfer handler is removed.
-        previewPane.setContentType("text/html");
-        previewPane.addHyperlinkListener(hyperlinkEvent -> {
-            if ((hyperlinkEvent.getEventType() == HyperlinkEvent.EventType.ACTIVATED) && PreviewPanel.this.databaseContext
-                    .isPresent()) {
-                try {
-                    String address = hyperlinkEvent.getURL().toString();
-                    JabRefDesktop.openExternalViewer(PreviewPanel.this.databaseContext.get(), address, FieldName.URL);
-                } catch (IOException e) {
-                    LOGGER.warn("Could not open external viewer", e);
+        addEventFilter(KeyEvent.KEY_PRESSED, event -> {
+            Optional<KeyBinding> keyBinding = Globals.getKeyPrefs().mapToKeyBinding(event);
+            if (keyBinding.isPresent()) {
+                switch (keyBinding.get()) {
+                    case COPY_PREVIEW:
+                        copyPreviewToClipBoard();
+                        event.consume();
+                        break;
+                    case CLOSE_DIALOG:
+                        close();
+                        event.consume();
+                        break;
+                    default:
                 }
             }
         });
+    }
 
+    private ContextMenu createPopupMenu() {
+        MenuItem copyPreview = new MenuItem(Localization.lang("Copy preview"), IconTheme.JabRefIcon.COPY.getGraphicNode());
+        copyPreview.setOnAction(event -> copyPreviewToClipBoard());
+        MenuItem printEntryPreview = new MenuItem(Localization.lang("Print entry preview"), IconTheme.JabRefIcon.PRINTED.getGraphicNode());
+        printEntryPreview.setOnAction(event -> print());
+        MenuItem previousPreviewLayout = new MenuItem(Localization.menuTitle("Previous preview layout"));
```

Comment:
On my machine, the localization string appears prefixed with an `&`. Even if that's not nicely solvable here, could you at least implement a quick hack to remove the preceeding `&`?

Also, in the Swing version the context menu text also listed the shortcuts for the menu item. This should be the case here as well.

Reply:
I removed the & and added the display of the shortcut.